extensions/wikia/Recirculation/js/views/mixedFooter.js

```diff
@@ -0,0 +1,79 @@
+define('ext.wikia.recirculation.views.mixedFooter', [
+	'jquery',
+	'wikia.window',
+	'ext.wikia.recirculation.tracker',
+	'ext.wikia.recirculation.utils'
+], function ($, w, tracker, utils) {
+	'use strict';
+
+	function render(data) {
+		var	newsAndStoriesList = data.nsItems.items,
+			wikiArticlesList = data.wikiItems.items,
+			templateList = getTemplateList(newsAndStoriesList),
+			templates = {};
+
+		$('.mcf-discussions-placeholder').replaceWith(data.discussions);
+
+		utils.loadTemplates(templateList)
+			.then(function (data) {
+				templateList.forEach(function (templateName, index) {
+					templates[templateName] = data[index];
+				});
+				injectTemplates(templates, newsAndStoriesList, wikiArticlesList);
+			});
+	}
+
+	function injectTemplates(templates, newsAndStoriesList, wikiArticlesList) {
+		var $newsAndStoriesHook = $('.ns-article'),
```

Comment:
I would rename this classes. Currently they are too ambiguous.
For example `.mcf-card-ns-placeholder` and `.mcf-card-wiki-placeholder`